crates/evm/src/execute.rs

```diff
@@ -287,7 +280,38 @@ where
     pub(crate) ctx: F::ExecutionCtx<'a>,
     pub(crate) parent: &'a SealedHeader<HeaderTy<N>>,
     pub(crate) assembler: Builder,
-    pub(crate) simulated: bool,
+}
+
+/// Conversions for executable transactions.
+pub trait ExecutorTx<B: BlockExecutor> {
+    /// The inner signed transaction.
+    type T;
```

Comment:
do we need the extra associated type here if we already have access to the executor type's `ExecutableTx` like

`ExecutableTx<Builder::Executor>`

which is what we're forwarding